src/Program.cs

```diff
@@ -229,41 +201,62 @@ private static int Pull(PullOptions opts)
 
                     opts.Version = project.Runtime.Version;
                     opts.Platform = project.Runtime.Platform;
-                    opts.Tier = project.Runtime.Tier == ServerTier.Pro;
+                    opts.Pro = project.Runtime.Tier == ServerTier.Pro;
                 }
                 else
                 {
-                    Console.Error.WriteLine(Output.Red($"You can perform this command only in a project directory."));
+                    Console.Error.WriteLine(Output.Red($"Couldn't find a version to install. To download latest version use option --latest"));
                     return 2;
                 }
             }
 
-            // if version provided is "latest", it is being replaced with currently most recent one
-            if (opts.Version == "latest")
-            {
-                opts.Version = VersionManager.GetLatestDarkRiftVersion();
-            }
+            ServerTier actualTier = opts.Pro ? ServerTier.Pro : ServerTier.Free;
 
             // If --docs was specified, download documentation instead
-            string path;
+            bool success = false;
             if (opts.Docs)
             {
-                path = VersionManager.GetDocumentationPath(opts.Version);
+                bool docsInstalled = VersionManager.IsDocumentationInstalled(opts.Version);
+
+                if (docsInstalled && !opts.Force)
+                {
+                    Console.WriteLine(Output.Green($"Documentation for DarkRift {opts.Version} - {actualTier} (.NET {opts.Platform}) already installed! To force a reinstall use the option -f or --force"));
+                    success = true;
+                }
+                else
+                    success = VersionManager.DownloadDocumentation(opts.Version);
             }
-            else
+            else if (opts.Version != null)
             {
-                path = VersionManager.GetInstallationPath(opts.Version, opts.Tier ? ServerTier.Pro : ServerTier.Free, opts.Platform);
+                bool versionInstalled = VersionManager.IsVersionInstalled(opts.Version, actualTier, opts.Platform);
+                if (versionInstalled && !opts.Force)
+                {
+                    Console.WriteLine(Output.Green($"DarkRift {opts.Version} - {actualTier} (.NET {opts.Platform}) already installed! To force a reinstall use the option -f or --force"));
+                    success = true;
+                }
+                else
+                    success = VersionManager.DownloadVersion(opts.Version, actualTier, opts.Platform);
             }
 
-            if (path == null)
+            if (!success)
```

Comment:
`success` being `false` doesn't always indicate the command is invalid. For example, if `DownloadDocumetation`/`DownloadVersion` fails due to internet connectivity.

I'm not actually sure I can see how `success` being `false` is ever an invalid command actually.